docs(Grid): add filter, sort, editing and group tools with pager adaptiveness (#2945)
telerik/blazor-docs: components/grid/toolbar.md

| Tool Name | Tool Tag | Description |
| --- | --- | --- |
| Add | `GridToolBarAddTool` | An add command that fires the [`OnAdd` event](slug:grid-editing-overview#events). |

> | Cancel || `GridToolBarCancelEditTool` | Cancels the changes for the selected row. Row selection and `Inline` or `Popup` editing mode are required. |

Reviewer: Don't leave cells blank. Enter "NA" in the cell if there is no Tool Tag. This comment applies to all blank cells.

| CsvExport | `GridToolBarCsvExportTool` | An export command for CSV files that fires the [`OnBeforeExport` event](slug:grid-export-events#onbeforeexport). |
| Delete || `GridToolBarDeleteTool` | Deletes the selected row. Row selection and `Inline` or `Popup` editing mode are required. |
| Edit || `GridToolBarEditTool` | Enters edit mode for the selected row. Row selection and `Inline` or `Popup` editing mode are required. |
| ExcelExport | `GridToolBarExcelExportTool` | An export command for Excel files that fires the [`OnBeforeExport` event](slug:grid-export-events#onbeforeexport). |

> | Filter | `GridToolBarFilterTool` | A toggle button in the Grid’s toolbar that opens a filter interface. On desktop screens, it displays a popup with a filter menu; on mobile devices, it renders as an `ActionSheet`. The component has two views: one for selecting the column to filter, and another for applying the filter to the selected column. The tool also exposes an `Icon` parameter that allows you to override the default icon.  |

Reviewer: ```suggestion
| Filter | `GridToolBarFilterTool` | A toggle button in the Grid’s toolbar that opens a filter interface. On desktop screens, it displays a popup with a filter menu; on mobile devices, it renders an `ActionSheet`. The filter component has two views: one for selecting the column to filter, and another for applying the filter to the selected column. The tool also exposes an `Icon` parameter that allows you to override the default icon.  |
```

> | Filter | `GridToolBarFilterTool` | A toggle button in the Grid’s toolbar that opens a filter interface. On desktop screens, it displays a popup with a filter menu; on mobile devices, it renders an `ActionSheet`. The filter component has two views: one for selecting the column to filter, and another for applying the filter to the selected column. The tool also exposes an `Icon` parameter that allows you to override the default icon.  |

Reviewer 2: ```suggestion
| Filter | `GridToolBarFilterTool` | A toggle button in the Grid’s toolbar that opens a UI option for filtering. On desktop screens, it displays a popup with a [filter menu]; on mobile devices, it renders an `ActionSheet`. The filter component has two views: one for selecting the column to filter, and another for applying the filter to the selected column. The tool also exposes an `Icon` parameter that allows you to override the default icon.  |
```

| Add | `GridToolBarAddTool` | An add command that fires the [`OnAdd` event](slug:grid-editing-overview#events). |
| CsvExport | `GridToolBarCsvExportTool` | An export command for CSV files that fires the [`OnBeforeExport` event](slug:grid-export-events#onbeforeexport). |
| ExcelExport | `GridToolBarExcelExportTool` | An export command for Excel files that fires the [`OnBeforeExport` event](slug:grid-export-events#onbeforeexport). |

> | Filter | `GridToolBarFilterTool` | A toggle button in the Grid’s toolbar that opens a filter interface. On desktop screens, it displays a popup with a filter menu; on mobile devices, it renders as an `ActionSheet`. The component has two views: one for selecting the column to filter, and another for applying the filter to the selected column. The tool also exposes `Icon` parameter that allows you to override the default icon.  |

Reviewer: ```suggestion
| Filter | `GridToolBarFilterTool` | A toggle button in the Grid’s toolbar that opens a filter interface. On desktop screens, it displays a popup with a filter menu; on mobile devices, it renders as an `ActionSheet`. The component has two views: one for selecting the column to filter, and another for applying the filter to the selected column. The tool also exposes an `Icon` parameter that allows you to override the default icon.  |
```

> | Group | `GridToolBarGroupTool` | A toggle button in the Grid’s toolbar that opens a popup listing the groupable columns — click a column to group by it. On mobile devices, the popup is rendered as an `ActionSheet`. The tool also exposes `Icon` parameter that allows you to override the default icon. |

Reviewer: Our style guide recommends using the em dash without spaces on either side.
```suggestion
| Group | `GridToolBarGroupTool` | A toggle button in the Grid’s toolbar that opens a popup listing the groupable columns—click a column to group by it. On mobile devices, the popup is rendered as an `ActionSheet`. The tool also exposes an `Icon` parameter that allows you to override the default icon. |
```

> | Sort | `GridToolBarSortTool` | A toggle button in the Grid’s toolbar that opens a popup listing the sortable columns — click a column to sort by it. On mobile devices, the popup is rendered as an `ActionSheet`. The tool also exposes `Icon` parameter that allows you to override the default icon. |

Reviewer: ```suggestion
| Sort | `GridToolBarSortTool` | A toggle button in the Grid’s toolbar that opens a popup listing the sortable columns—click a column to sort by it. On mobile devices, the popup is rendered as an `ActionSheet`. The tool also exposes an `Icon` parameter that allows you to override the default icon. |
```

            Export to Excel
        </GridToolBarExcelExportTool>

        <GridToolBarFilterTool>
            Filter
        </GridToolBarFilterTool>

        <GridToolBarSortTool>
            Sort
        </GridToolBarSortTool>

        <GridToolBarGroupTool>
            Group
        </GridToolBarGroupTool>

>         <GridToolBarAddTool>

Reviewer 2: This is a second instance of the Add tool. I believe we don't need that twice.
